config/base/src/lib.rs

```diff
@@ -256,3 +271,59 @@ pub mod view {
         pub const IS_HAS_VIEW: bool = true;
     }
 }
+
+pub mod proxy {
+    //! Module for configuration proxies' traits
+
+    use super::*;
+
+    /// Pseudo-default trait only used for doc generation
+    pub trait DocsDefault {
```

Comment:
Which is very much the exact same thing as `Default`, can be used in the same way, and people will probably abuse in the same way. 

We need to get rid not only of `core::Default` but also of any trait that can be used like it. Adding a `DocsDefault` trait in the exact same spot doesn't solve anything.